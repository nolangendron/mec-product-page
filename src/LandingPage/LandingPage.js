import React from 'react'
import styled from '@emotion/styled'
import bgImage from '../images/bgImage.jpg'
import logo from '../images/MEC_logo_2013.svg'
import { SearchInput } from '../components'
import { Redirect } from 'react-router-dom'

const LandingPageStyles = styled('div')`
  height: 100vh;
  width: 100vw;

  .hero-image {
    background: url(${bgImage}) no-repeat;
    background-position: center;
    background-size: cover;
    background-attachment: fixed;
    height: 70%;
  }

  .logo {
    text-align: center;
    margin-top: -87.5px;
  }

  .logo img {
    width: 175px;
    height: 175px;
  }
`

export const LandingPage = ({
  currentSearch,
  searchQuery,
  handleQueryChange,
  handleSearch,
  redirect,
  clearSearch,
}) => {
  return (
    <LandingPageStyles>
      <div className="hero-image"></div>
      <div className="logo">
        <img src={logo} alt="MEC logo" />
      </div>
      <SearchInput
        landingPage
        searchQuery={searchQuery}
        handleQueryChange={handleQueryChange}
        handleSearch={handleSearch}
        clearSearch={clearSearch}
      />
      {redirect && <Redirect push to={`/products`} />}
    </LandingPageStyles>
  )
}
