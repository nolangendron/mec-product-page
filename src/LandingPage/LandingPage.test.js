import React from "react";
import { render } from "@testing-library/react";
import { LandingPage } from "./LandingPage";

it("should take a snapshot of LandingPage", () => {
  const { asFragment } = render(<LandingPage />);
  expect(asFragment(<LandingPage />)).toMatchSnapshot();
});
