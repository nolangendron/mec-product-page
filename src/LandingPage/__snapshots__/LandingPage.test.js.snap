// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot of App 1`] = `
<DocumentFragment>
  <div>
    Landing Page
  </div>
</DocumentFragment>
`;
